src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -169,40 +146,153 @@ static inline ucc_status_t ucc_tl_mlx5_a2a_team_test(ucc_base_team_t *team)
 
 ucc_status_t ucc_tl_mlx5_team_create_test(ucc_base_team_t *team)
 {
-    ucc_tl_mlx5_team_t *tl_team      = ucc_derived_of(team, ucc_tl_mlx5_team_t);
-    ucc_status_t        a2a_status   = UCC_OK;
-    ucc_status_t        mcast_status = UCC_OK;
+    ucc_tl_mlx5_team_t            *tl_team      = ucc_derived_of(team, ucc_tl_mlx5_team_t);
+    ucc_team_t                    *core_team    = UCC_TL_CORE_TEAM(tl_team);
+    ucc_subset_t                   subset       = {.map = UCC_TL_TEAM_MAP(tl_team),
+                                                .myrank = UCC_TL_TEAM_RANK(tl_team)};
+    ucc_status_t                   a2a_status   = UCC_OK;
+    ucc_status_t                   mcast_status = UCC_OK;
+    ucc_tl_mlx5_mcast_coll_comm_t *comm         = NULL;
+    ucc_status_t                   status;
+
+
+    if (tl_team->global_sync_in_progress) {
+        status = ucc_service_coll_test(tl_team->global_sync_req);
+        if (status < 0) {
+            tl_debug(UCC_TL_TEAM_LIB(tl_team),
+                     "failure during service coll exchange: %s",
+                     ucc_status_string(status));
+            return status;
+        }
+        if (UCC_INPROGRESS == status) {
+            return status;
+        }
+
+        ucc_service_coll_finalize(tl_team->global_sync_req);
+
+        tl_team->global_sync_in_progress = 0;
+
+        if (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_CTX_CHECK &&
+            tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_CTX_CHECK ) {
+            tl_team->a2a_status.global = tl_team->global_status_array[UCC_TL_MLX5_A2A_STATUS_INDEX];
+            tl_team->a2a_state         = TL_MLX5_TEAM_STATE_ALLTOALL_INIT;
```

Comment:
why checking only mcast status and not alltoall?

Reply:
a2a is being handled separately somewhere else. regardless of the global status for a2a, we must set its state to TL_MLX5_TEAM_STATE_ALLTOALL_INIT and let it get handled under a2a calls.